src/MandrillServiceProvider.php

```diff
@@ -6,12 +6,13 @@
 
 class MandrillServiceProvider extends ServiceProvider
 {
-    public function boot()
+    public function register()
     {
-        $this->app['mail.manager']->extend('mandrill', function () {
-            $config = $this->app['config']->get('services.mandrill', []);
-
-            return new MandrillTransport(new \GuzzleHttp\Client($config), $config['secret']);
+        $this->app->resolving(\Illuminate\Mail\MailManager::class, function (\Illuminate\Mail\MailManager $mail_manager) {
+            $mail_manager->extend("mandrill", function () {
+                $config = $this->app['config']->get('services.mandrill', []);
```

Comment:
We should probably use `afterResolving()` here so we are sure the instance was resolved from the container before trying to extend it.

Please import the namespaces vs. inlining them.